File: backend/src/main/scala/microservices/content/objects/ReviewPlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ReviewPlannerResponse(
    reviewId: String,
    authorUserId: String,
    authorDisplayName: String,
    authorAvatarUrl: Option[String],
    resourceType: String,
    resourceId: String,
    resourceSummaryTitle: String,
    resourceSummarySubtitle: String,
    orderId: String,
    orderItemId: String,
    rating: Int,
    title: String,
    content: String,
    status: String,
    createdAt: String,
    updatedAt: String,
    isMyReview: Boolean,
    canEdit: Boolean,
    canDelete: Boolean,
    images: List[ContentImagePlannerResponse]
)
object ReviewPlannerResponse:
  given sourceEncoder: Encoder[ReviewPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ReviewPlannerResponse] = deriveDecoder

File: frontend/src/microservices/content/objects/ReviewPlannerResponse.ts
// ReviewPlannerResponse：content 域单条评论返回对象。

import type { ContentImagePlannerResponse } from './ContentImagePlannerResponse'
export type ReviewPlannerResponse = {
  reviewId: string
  authorUserId: string
  authorDisplayName: string
  authorAvatarUrl: string | null
  resourceType: string
  resourceId: string
  resourceSummaryTitle: string
  resourceSummarySubtitle: string
  orderId: string
  orderItemId: string
  rating: number
  title: string
  content: string
  status: string
  createdAt: string
  updatedAt: string
  isMyReview: boolean
  canEdit: boolean
  canDelete: boolean
  images: ContentImagePlannerResponse[]
}
export const reviewPlannerResponseFromJson = (json: string): ReviewPlannerResponse =>
  JSON.parse(json) as ReviewPlannerResponse
export const reviewPlannerResponseToJson = (value: ReviewPlannerResponse): string =>
  JSON.stringify(value)
